src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.RegularCompilationTracker.cs

```diff
@@ -612,7 +612,7 @@ await compilationState.GetCompilationAsync(
                 if (workspacePreference != SourceGeneratorExecutionPreference.Automatic)
                 {
                     if (creationPolicy.GeneratedDocumentCreationPolicy == GeneratedDocumentCreationPolicy.Create)
-                        creationPolicy = creationPolicy with { GeneratedDocumentCreationPolicy = GeneratedDocumentCreationPolicy.DoNotCreate };
+                        creationPolicy = creationPolicy with { GeneratedDocumentCreationPolicy = GeneratedDocumentCreationPolicy.CreateRequired };
```

Comment:
I was going to complain that i didn't write docs here, and that was my bad.  But turns out i did, and they're great :p

So please update the docs above this appropriately.  It should be expanded to indicate what this new mode is, and how it affects 'required' and non-required sgs.

--

aside, as i talk about this more, my brain keeps referring to these generators as 'critical' vs 'required'.  perhaps that name might be better.  but i' mnot pushing for that.